tests/nativeapp/test_run_processor.py

```diff
@@ -184,29 +198,78 @@ def test_create_dev_app_create_new_w_no_additional_privileges(
             (None, mock.call("use role old_role")),
         ]
     )
-    mock_conn.return_value = MockConnectionCtx()
-    mock_execute.side_effect = side_effects
+    mock_execute_query.side_effect = side_effects
 
-    mock_diff_result = DiffResult()
     current_working_directory = os.getcwd()
     create_named_file(
         file_name="snowflake.yml",
         dir_name=current_working_directory,
-        contents=[mock_snowflake_yml_file.replace("package_role", "app_role")],
+        contents=[mock_snowflake_yml_file],
     )
 
     run_processor = _get_na_run_processor()
-    assert not mock_diff_result.has_changes()
     run_processor.create_or_upgrade_app(
         policy=MagicMock(), install_method=SameAccountInstallMethod.unversioned_dev()
     )
+    assert mock_execute_query.mock_calls == mock_execute_query_expected
+
+
+# Test create_dev_app with exception thrown trying to use the warehouse
+@mock.patch(NATIVEAPP_MANAGER_EXECUTE)
+@mock.patch(NATIVEAPP_MANAGER_USE_ROLE)
+@mock_connection()
+def test_create_dev_app_w_warehouse_access_exception(
+    mock_conn, mock_use_role, mock_execute, temp_dir, mock_cursor
+):
```

Comment:
In this test and the one below, we mock use_role with`@mock.patch(NATIVEAPP_MANAGER_USE_ROLE)`. The mocking of `_execute_query` that was made as part of use_role are removed from `mock_execute_helper`.